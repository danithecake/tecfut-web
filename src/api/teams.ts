import type { EndpointOutput } from '@sveltejs/kit'
import { getTeams } from 'lib/teams'

export const get: () => Promise<EndpointOutput> = () => {
  return new Promise((resolve) => void resolve({ body: JSON.stringify(getTeams()) }))
}
