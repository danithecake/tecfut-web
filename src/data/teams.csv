name,fullname
blue,Синие
manki,Манишки
