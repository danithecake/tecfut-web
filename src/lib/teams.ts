import type { ITeam } from 'common/interfaces'
import fs from 'fs'
import path from 'path'
import Papa from 'papaparse'
import { DATA_DIR } from 'common/constants'

const PAPA_CONF: Papa.ParseConfig<ITeam> = {
  header: true,
  dynamicTyping: true,
  skipEmptyLines: true
}

type PlayerName = ITeam['name']

export const getTeams: () => Record<PlayerName, ITeam> = () => {
  try {
    return Papa.parse(fs.readFileSync(path.join(DATA_DIR, 'teams.csv'), 'utf8'), PAPA_CONF).data.reduce(
      (teams, team) => {
        teams[team.name] = team

        return teams
      },
      {} as Record<PlayerName, ITeam>
    )
  } catch (err) {
    console.error(err)

    return {}
  }
}
