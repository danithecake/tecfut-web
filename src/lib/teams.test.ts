import { mocked } from 'ts-jest/utils'
import Papa from 'papaparse'
import { getTeams } from './teams'

describe('lib/teams', () => {
  describe('getPlayers', () => {
    test('returns parsed teams map from local teams.csv file', () => {
      expect(getTeams()).toMatchSnapshot()
    })

    test('returns empty teams map and prints error to console if something went wrong', () => {
      jest.spyOn(Papa, 'parse').mockImplementationOnce(() => {
        throw new Error('Something went wrong')
      })
      jest.spyOn(console, 'error')

      expect(getTeams()).toEqual({})
      expect(mocked(console.error)).toBeCalledWith(expect.any(Error))

      mocked(Papa.parse).mockRestore()
      mocked(console.error).mockRestore()
    })
  })
})
