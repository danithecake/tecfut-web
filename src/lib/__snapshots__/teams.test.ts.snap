// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/teams getPlayers returns parsed teams map from local teams.csv file 1`] = `
Object {
  "blue": Object {
    "fullname": "Синие",
    "name": "blue",
  },
  "manki": Object {
    "fullname": "Манишки",
    "name": "manki",
  },
}
`;
